src/storage/src/internal_control.rs

```diff
@@ -55,12 +55,21 @@ pub enum InternalStorageCommand {
 pub trait InternalCommandSender {
     /// Broadcasts the given command to all workers.
     fn broadcast(&mut self, internal_cmd: InternalStorageCommand);
+
+    /// Returns the next available command, if any. This returns `None` when
+    /// there are currently no commands but there might be commands again in the
+    /// future.
+    fn next(&mut self) -> Option<InternalStorageCommand>;
```

Comment:
I don't like exposing it like this, but I think we have some more work to do if we want to make the abstraction tighter, so this seems fine for now. But, idk ... :man_shrugging:

Reply:
Maybe I'm a little slow on Monday morning, but can you point out where this method is used / why it's needed?

Reply:
It's used roughly here: https://github.com/MaterializeInc/materialize/blob/7db86cf650a072570891f3288fdd01db5b926f43/src/storage/src/storage_state.rs#L417

Reply:
Ah, okay. To spell it out a bit more: the type of `command_sequencer` in `run_client` below has weakened; we used to always have a specific type of sequencer but now we have a `dyn` of the shared trait. And next was only available for the specific type.

Annoying but reasonable I think!

Reply:
> Annoying but reasonable I think!

yeah, annoying ... :sweat_smile: And yes, your spelling out was correct!